### PR TITLE
Feat/localcache

### PR DESCRIPTION
**Pull request by MaxC-AT**

### CODE REVIEW

File: localcache/localcache.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+// localCache is a simple lru cache implementation with ttl.
+type localCache struct {
+		m        sync.Mutex
+		ttl      time.Duration
+		cap      int
+		size     int
+		store    map[string]*list.Element
+		lrulist  *list.List
+		exppq		 priorityQueue
```
Comment (VineyAT) [inline localcache/localcache.go]: It's better to run `linter` before committing.

File: localcache/localcache_test.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+)
+
+type localCacheSuite struct {
+		suite.Suite
+		localcache *localCache
+}
+
+func (s *localCacheSuite) SetupSuite() {}
+
+func (s *localCacheSuite) SetupTest() {}
```
Comment (VineyAT) [inline localcache/localcache_test.go]: 可以用 localcache.go 上面的 `New()` 產生新物件放在 localCacheSuite 底下

File: localcache/localcache_test.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+		}{
+				{ desc: "nil if key not existed", key: "test1", val: "test1", exp: time.Now().Add((mockTTL) * time.Second).UnixNano() },
+				{ desc: "nil if key had expired", key: "test", val: "test", exp: time.Now().UnixNano() },
+		}
+
+		for _, t := range tests {
+				items := createCacheItems(
+						[][]interface{}{{t.key, t.val, t.exp}},
+				)
+				s.localcache = mockNew(items)
```
Comment (VineyAT) [inline localcache/localcache_test.go]: 同上，你還額外刻了一個 mockNew，這樣你怎確保實際用的 New 跟 mockNew 之間的關聯？應該都從 New 開始，才能確保內部使用跟外部使用一致

File: localcache/localcache_test.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+				val, ok := s.localcache.Get("test")
+				s.Require().Nil(val)
+				s.Require().False(ok)
+		}
+}
+
+func (s *localCacheSuite) TestGetValue() {
```
Comment (VineyAT) [inline localcache/localcache_test.go]: `TestGet` -> 跟 interface 一致，`TestSet` 我就不寫了

File: localcache/localcache.go (hunk 1 above)
Comment (MaxC-AT) [reply localcache/localcache.go]: Fixed

File: localcache/localcache_test.go (hunk 3 above)
Comment (MaxC-AT) [reply localcache/localcache_test.go]: Fixed

File: localcache/localcache_test.go (hunk 4 above)
Comment (MaxC-AT) [reply localcache/localcache_test.go]: Fixed

File: localcache/localcache_test.go (hunk 2 above)
Comment (MaxC-AT) [reply localcache/localcache_test.go:25]: As there are multiple test cases within a single test function, I couldn't include 's.localcache = New()' in the SetupTest function. Instead, I will create a new local cache object before running each test case.

Comment (MaxC-AT) [reply localcache/localcache_test.go]: Fixed

**Pull request merged**